src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,230 @@
-        catalog["separation"] = separations
-        catalog = catalog[separations < radius]
 
-    catalog.remove_columns(["RAJ2000", "DEJ2000"])
+    return stars_
+
+
+def get_star_catalog(
+    catalog: str | StarCatalog, magnitude_cutoff: float = 8.0, row_limit: int = 1000000
+) -> Table:
+    """
+    Utility function to download a star catalog for the get_bright_stars function.
+
+    Parameters
+    ----------
+    catalog : str or ctapipe.utils.astro.StarCatalog
+        Name of the catalog to be used. Usable names are found in the Enum StarCatalog. Default: Yale.
+    magnitude_cutoff : float, optional
+        Maximum value for magnitude used in lookup. Default is 8.0.
+    row_limit : int, optional
+        Maximum number of rows for the star catalog lookup. Default is 1000000.
+
+    Returns
+    -------
+    astropy.table.Table
+        List of all stars after cuts with catalog numbers, magnitudes,
+        and coordinates as SkyCoord objects including proper motion.
+    """
+    from astroquery.vizier import Vizier
+
+    if isinstance(catalog, str):
+        catalog = StarCatalog[catalog]
+    catalog_dict = catalog.value
+
+    vizier = Vizier(
+        catalog=catalog_dict.directory,
+        columns=catalog_dict.columns,
+        row_limit=row_limit,
+    )
+
+    stars = vizier.query_constraints(Vmag=f"<{magnitude_cutoff}")[0]
+    meta = catalog_dict._asdict()
+    meta["magnitude_cutoff"] = magnitude_cutoff
+
+    stars.meta["Catalog"] = meta
+
+    return stars
+
+
+def get_bright_stars(
+    time: Time,
+    catalog: StarCatalog | str = "Yale",
+    pointing: SkyCoord | None = None,
+    radius: Quantity | None = None,
+    magnitude_cut: float | None = None,
+) -> Table:
+    """
+    Get an astropy table of bright stars from the specified star catalog.
+
+    Parameters
+    ----------
+    time : astropy.time.Time
+        Time to which proper motion is applied.
+    catalog : str or ctapipe.utils.astro.StarCatalog, optional
+        Name of the catalog to be used. Available catalogues are 'Yale' and 'Hipparcos'. Default is 'Yale'.
+    pointing : astropy.coordinates.SkyCoord, optional
+        Pointing direction in the sky. If None is given, the full sky is returned.
+    radius : astropy.units.Quantity, optional
+        Radius of the sky region around the pointing position. Default is the full sky.
+    magnitude_cut : float, optional
+        Return only stars above a given absolute magnitude. Default is None (all entries).
+
+    Returns
+    -------
+    astropy.table.Table
+        List of all stars after applying the cuts, with catalog numbers, magnitudes,
+        and coordinates as SkyCoord objects including proper motion.
+    """
+    from importlib.resources import files
+
+    if isinstance(catalog, str):
+        catalog = StarCatalog[catalog]
+    cat = catalog.value
+    record = cat.record
+
+    f = files("ctapipe").joinpath(f"resources/{record}.fits.gz")
```

Comment:
Proper usage of the importlib API (as shown in the example I posted) is:

```
with as_file(files("ctapipe").joinpath(f"resources/{record}.fits.gz")) as f:
    stars = Table.read(f)
```

Reply:
This guarantees that the resource file is actually available as a normal file on disk, which isn't the case when the source module is zipped.

Reply:
I had this before and got an error saying that the output of files and as_files is depricated for use as context. I will reproduce it in a moment

Reply:
That's strange, there is no mention of any deprecations in the current documentation:
https://docs.python.org/3/library/importlib.resources.html#importlib.resources.as_file

Reply:
I just changed it locally and don't see any deprecation either...

Reply:
re-installed my cta-dev environment and it is now gone. I will implement this